tests/components/modbus/conftest.py

```diff
@@ -118,8 +122,24 @@ async def base_test(
             async_fire_time_changed(hass, now)
             await hass.async_block_till_done()
 
-        # Check state
         entity_id = f"{entity_domain}.{device_name}"
+
+        # Call an arbitrary service
+        if "call_service" in kvargs:
```

Comment:
Looks like you are allowing both a read and a write test in the same call, the idea of this function is one call pr test.

Reply:
Yeah, this test design might be confusing. Let me rethink it based on the cache decision